src/main/java/ch/njol/skript/lang/Variable.java

```diff
@@ -294,14 +295,20 @@ public <R> Variable<R> getConvertedExpression(Class<R>... to) {
 	 * Gets the value of this variable as stored in the variables map.
 	 */
 	@Nullable
-	public Object getRaw(Event e) {
-		String n = name.toString(e);
-		if (n.endsWith(Variable.SEPARATOR + "*") != list) // prevents e.g. {%expr%} where "%expr%" ends with "::*" from returning a Map
+	public Object getRaw(Event event) {
+		String name = this.name.toString(event);
+		if (name.endsWith(Variable.SEPARATOR + "*") != list) // prevents e.g. {%expr%} where "%expr%" ends with "::*" from returning a Map
 			return null;
-		Object val = !list ? convertIfOldPlayer(n, e, Variables.getVariable(n, e, local)) : Variables.getVariable(n, e, local);
-		if (val == null)
-			return Variables.getVariable((local ? LOCAL_VARIABLE_TOKEN : "") + name.getDefaultVariableName(), e, false);
-		return val;
+		Object value = !list ? convertIfOldPlayer(name, event, Variables.getVariable(name, event, local)) : Variables.getVariable(name, event, local);
+		// Check for default variables.
```

Comment:
Like a `isDefaultVariable()` type method? How would you add this to variables? Also I don't see a use for letting the add-on know about default variables. The add-on just needs to value.

Reply:
My original question was wrong, I didn't think right about default variables.

An issue I see is that an addon may have some syntax related to getting a variable. For example, when a script has default default variable `{stuff::%player%}`, and an addon tries to get the variable `{stuff::TPGamesNL}`, this will not use the default value if needed, because `Variables#getVariable` takes a string for the variable name, without context of the types of parts of the string. 

We should probably add a note to the Javadocs of `Variables` that it is preferred to use methods from `Variable` when using variables from user input in scripts, because they do take default variables into account.